components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1148,13 +1135,8 @@ void RewardsServiceImpl::OnAttestationAndroid(
     const bool token_received,
     const std::string& token,
     const bool attestation_passed) {
-  if (!Connected()) {
-    return;
-  }
-
-  if (!token_received) {
+  if (!Connected() || !token_received) {
```

Comment:
Shouldn't `std::move(callback).Run(ledger::mojom::Result::LEDGER_ERROR, nullptr);` be deferred as well?

Reply:
This method is itself a callback, so I'm fairly certain it will already be executing asynchronously.